### PR TITLE
feat: eth address signin

### PR DESCRIPTION
**Pull request by saitofun**

### CODE REVIEW

File: cmd/srv-applet-mgr/main.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func main() {
 				if err := deploy.StartInstances(
 					global.WithContext(context.Background()),
 				); err != nil {
-					panic(err)
+					// panic(err)
```
Comment (CoderZhi) [inline cmd/srv-applet-mgr/main.go:49]: add a `TODO`

**Review by CoderZhi — COMMENTED** [review]
change all `eth` to `wallet`

**Review by CoderZhi — APPROVED** [review]

File: cmd/srv-applet-mgr/apis/account/root.go [hunk 2]
```diff
@@ -5,9 +5,14 @@ import (
...
+	Root         = kit.NewRouter(httptransport.Group("/account"))
+	RegisterRoot = kit.NewRouter(httptransport.Group("/register"))
+)
 
 func init() {
-	Root.Register(kit.NewRouter(&CreateAccount{}))
+	Root.Register(kit.NewRouter(&CreateAccountByUsernameAndPassword{}))
 	Root.Register(kit.NewRouter(&UpdatePasswordByAccountID{}))
+
+	RegisterRoot.Register(kit.NewRouter(&CreateAccountByUsernameAndPassword{}))
```
Comment (Liuhaai) [inline cmd/srv-applet-mgr/apis/account/root.go:17]: duplicate on L14 and L17?

File: cmd/srv-applet-mgr/apis/account/root.go [hunk 3]
```diff
@@ -5,9 +5,14 @@ import (
 	"github.com/machinefi/w3bstream/pkg/depends/kit/kit"
 )
 
-var Root = kit.NewRouter(httptransport.Group("/account"))
+var (
+	Root         = kit.NewRouter(httptransport.Group("/account"))
+	RegisterRoot = kit.NewRouter(httptransport.Group("/register"))
```
Comment (Liuhaai) [inline cmd/srv-applet-mgr/apis/account/root.go:10]: what's the purpose for `/register` and `/account`?

File: cmd/srv-applet-mgr/global/config.go [hunk 4]
```diff
@@ -89,7 +89,8 @@ func init() {
 		types.WithWasmDBExecutorContext(config.WasmDB),
 		types.WithPgEndpointContext(config.Postgres),
 		types.WithRedisEndpointContext(config.Redis),
-		types.WithLoggerContext(conflog.Std()),
+		types.WithLoggerContext(config.StdLogger),
+		conflog.WithLoggerContext(config.StdLogger),
```
Comment (Liuhaai) [inline cmd/srv-applet-mgr/global/config.go:93]: what's the differences between two contexts?

File: pkg/enums/account_source.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package enums
+
+//go:generate toolkit gen enum AccountSource
+type AccountSource uint8
+
+const (
+	ACCOUNT_SOURCE_UNKNOWN AccountSource = iota
+	ACCOUNT_SOURCE__INIT
```
Comment (Liuhaai) [inline pkg/enums/account_source.go:8]: what's the purpose of the enum

File: cmd/srv-applet-mgr/apis/login/put.go [hunk 6]
```diff
@@ -4,49 +4,57 @@ import (
...
+type LoginByUsername struct {
 	httpx.MethodPut
-	InBody `in:"body"`
+	account.LoginByUsernameReq `in:"body"`
 }
 
-type InBody struct {
-	Username string `json:"username"`
-	Password string `json:"password"`
+func (r *LoginByUsername) Path() string { return "" }
```
Comment (Liuhaai) [inline cmd/srv-applet-mgr/apis/login/put.go:20]: `/admin`?

File: pkg/models/account.go [hunk 7]
```diff
@@ -25,43 +22,8 @@ type RelAccount struct {
...
-func (AccountVendor) DataType(drv string) string { return "text" }
-
-func (v AccountVendor) Value() (driver.Value, error) {
-	return datatypes.JSONValue(v)
-}
-
-func (v *AccountVendor) Scan(src interface{}) error {
-	return datatypes.JSONScan(src, v)
+	Role   enums.AccountRole  `db:"f_role"              json:"role"`
+	State  enums.AccountState `db:"f_state,default='1'" json:"state"`
```
Comment (Liuhaai) [inline pkg/models/account.go:26]: what's purpose of the state of an account? allowing for a zombie account？

File: pkg/models/account_identity.go [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+//
+//go:generate toolkit gen model AccountIdentity --database DB
+type AccountIdentity struct {
+	datatypes.PrimaryID
+	RelAccount
+	AccountIdentityInfo
+	datatypes.OperationTimesWithDeleted
+}
+
+type AccountIdentityInfo struct {
```
Comment (Liuhaai) [inline pkg/models/account_identity.go:23]: what's this for?

File: cmd/srv-applet-mgr/main.go (hunk 1 above)
Comment (saitofun) [reply cmd/srv-applet-mgr/main.go:49]: fixed

File: cmd/srv-applet-mgr/apis/login/put.go [hunk 9]
```diff
@@ -4,49 +4,57 @@ import (
...
-	Token     string         `json:"token"`
-	ExpireAt  base.Timestamp `json:"expireAt"`
-	Issuer    string         `json:"issuer"`
+type LoginByEthAddress struct {
+	httpx.MethodPut
+	account.LoginByEthAddressReq `in:"body"`
 }
 
-func (r *Login) Path() string { return "/login" }
+func (r *LoginByEthAddress) Path() string { return "/eth" }
```
[the remark this answers is not in the document]
Comment (saitofun) [reply cmd/srv-applet-mgr/apis/login/put.go:35]: fixed

File: cmd/srv-applet-mgr/apis/account/root.go (hunk 2 above)
Comment (saitofun) [reply cmd/srv-applet-mgr/apis/account/root.go:17]: fixed

File: cmd/srv-applet-mgr/apis/account/root.go (hunk 3 above)
Comment (saitofun) [reply cmd/srv-applet-mgr/apis/account/root.go:10]: `/account` need token `/register` not

File: cmd/srv-applet-mgr/global/config.go (hunk 4 above)
Comment (saitofun) [reply cmd/srv-applet-mgr/global/config.go:93]: `conflog.FromContext` can get logger from current context, it return default logger `discard`, `types.FromContext` need assertion.

File: pkg/enums/account_source.go (hunk 5 above)
Comment (saitofun) [reply pkg/enums/account_source.go:8]: not always

File: cmd/srv-applet-mgr/apis/login/put.go (hunk 6 above)
Comment (saitofun) [reply cmd/srv-applet-mgr/apis/login/put.go:20]: this router is registered to `/login` all username login router

File: pkg/models/account.go (hunk 7 above)
Comment (saitofun) [reply pkg/models/account.go:26]: admin can remove or disable account manually

**Pull request merged**